core/eds.go

```diff
@@ -20,15 +19,15 @@ import (
 // ExtendedDataSquare (EDS). If there are no transactions in the block,
 // nil is returned in place of the eds.
 func extendBlock(data types.Data) (*rsmt2d.ExtendedDataSquare, error) {
-	if len(data.Txs) == 0 {
+	if len(data.Txs) == 0 && data.SquareSize == uint64(1) {
 		return nil, nil
 	}
 	shares, err := appshares.Split(data, true)
 	if err != nil {
 		return nil, err
 	}
-	size := utils.SquareSize(len(shares))
-	return da.ExtendShares(size, appshares.ToBytes(shares))
+
+	return da.ExtendShares(data.SquareSize, appshares.ToBytes(shares))
```

Comment:
I think I was wrong to say that this is necessary, but this is a good change anyway imo

Reply:
Yeah I think these should always be the same number. But better probably to use `data.SquareSize` from a defensive perspective